src/fidesops/models/privacy_request.py

```diff
@@ -131,6 +131,16 @@ class PrivacyRequest(Base):
         primaryjoin="foreign(ExecutionLog.privacy_request_id)==PrivacyRequest.id",
     )
 
+    # passive_deletes="all" prevents execution logs from having their privacy_request_id set to null when
```

Comment:
this is really cool! Didn't know we had this option to prevent foreign keys to be set to Null in cascade deletes. Could we get a unit test added to confirm this specific behavior?

Reply:
I didn't know about this option either before I started this ticket. I actually copied this from some code that I'm pretty sure Dawn wrote. One interesting thing to note is this relationship doesn't actually link to a foreign key. The new `audit_logs` relationship is linked to a normal column that is indexed. This is also how the `execution_logs` are done. I'm not 100% sure for the reasoning behind this but I based the new audit log table on the execution logs so I went with it. Maybe @pattisdr could weigh in on the reason for it not being a foreign key? 

Here is the column definition on the `ExecutionLog` Table as an example.
```python
    privacy_request_id = Column(
        String,
        nullable=False,
        index=True,
    )
```

Reply:
hmm, I originally implemented this as a FK, but this caused issues once Steven actually started connecting Graph Tasks to write to execution logs - back in October 2021?  I can't recall specifically what he was running into.

Even in the design review, both Neville and Steven had discouraged FK's here to write execution logs with as little fuss as possible.  Worth noting that privacy requests generate a ton of execution logs on each run. Every individual task in the graph gets an ExecutionLog for every single state.

Reply:
Thanks for the clarification!